app/controllers/users_controller.rb

```diff
@@ -0,0 +1,54 @@
+require 'pry'
+
+class UsersController < ApplicationController
+  get '/signup' do
+    if !session[:user_id]
+      erb :"users/create_user"
+    else
+      redirect '/tweets'
+    end
+  end
+
+  get "/users/:slug" do
+    @user = User.find_by_slug(params[:slug])
+    redirect "/tweets/#{user.slug}"
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:password] == "" || params[:email] ==""
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.